colin/core/checks/labels.py

```diff
@@ -73,3 +77,43 @@ def check(self, target):
                            reference_url=self.reference_url,
                            check_name=self.name,
                            logs=[])
+
+
+class InheritedOptionalLabelAbstractCheck(ImageAbstractCheck):
+
+    def __init__(self, message, description, reference_url, tags):
+        """
+        Abstract check for Dockerfile/Image labels.
+
+        :param message: str
+        :param description: str
+        :param reference_url: str
+        :param tags: [str]
+        """
+        super(InheritedOptionalLabelAbstractCheck, self) \
+            .__init__(message, description, reference_url, tags)
+
+    def check(self, target):
+        optional_labels = [
+            'summary',
+            'description',
+            'io.k8s.description',
+            'io.k8s.display-name',
+            'io.openshift.tags',
+        ]
+        passed = True
+
+        if target.parent_target:
+            labels_to_check = (set(optional_labels) & set(target.labels)
```

Comment:
I don't follow why the variable is named optional when it seems they are required actually since you are doing `optional AND child_labels AND parent_labels`

Reply:
well those are optional labels, but we first need to collect all which are present in both image and parent image, and then we compare values